test/common/http/http1/conn_pool_test.cc

```diff
@@ -537,7 +551,7 @@ TEST_F(Http1ConnPoolImplTest, ConnectionCloseWithoutHeader) {
   conn_pool_.test_clients_[0].connection_->raiseEvent(Network::ConnectionEvent::Connected);
 
   callbacks2.outer_encoder_->encodeHeaders(TestHeaderMapImpl{}, true);
-  response_headers.reset(new TestHeaderMapImpl{{":status", "200"}});
+  response_headers.reset(new TestHeaderMapImpl{{":status", "200"}, {"connection", "keep-alive"}});
```

Comment:
Do we need to change these tests?  If so, I'm a bit concerned the behavioral change might be larger scope than we intend.

Reply:
Unfortunately the tests do not specify HTTP/1.1 and assume that the lack of a Connection: close means keep-alive.  We could change them to be HTTP/1.1 or add the keep-alive. The protocol isn't explicitly specified so it might just be an oversight and in any case the protocol will be explicit in non-test operation.   Suggestion?

Reply:
Could default mock out the codec to return something other than 0 for protocol?  If that's not trivial I'm OK with this as-is - poked through the code and sorted out why it's the case.  As long as we didn't need to tweak integration tests I'm still happy it's low impact.

Reply:
The tests run at a higher layer (decodeHeaders) rather than running the parser_, so I would have to just override the protocol() which would mean that the behavior would not match any underlying data and the CodecClient::protocol() method is not virtual currently.  Also that would mean changing the test from HTTP/1.0 to HTTP/1.1.   I agree that it would be better to have these tests for both HTTP/1.0 and HTTP/1.1 and that the author might have thought that they were for 1.1 but they have been 1.0 so perhaps these should stay 1.0 and we can add some more 1.1 tests. 

Created https://github.com/envoyproxy/envoy/issues/7158

Reply:
+1 please don't change the existing tests, or clearly mark them as HTTP/1.0 and we can add HTTP/1.1 tests. Let's figure out how to write new tests that cover the HTTP/1.0 case, so let's resolve the issue you opened as part of this PR.

Reply:
I renamed the tests which depend on HTTP/1.0 behavior to include Http10 as a prefix.

Reply:
I think Matt's ask was that we either have 1.0 tests and 1.1 tests, or simply 1.1 tests, and while you did the rename we don't have the HTTP/1.1 case covered

So you can either copy/paste/edit the 1.0 tests to 1.1 tests, or go for minimal changes, say making expectClientCreate(Protocol protocol =  Protocol::Http11) ) (in your new test have it be HTTP/1.0) and adding
ON_CALL(test_client.codec_, protocol()).WillByDefault(Return(protocol)) to expectClientCreate

Reply:
Done.